pipeline/splicing/calcVarPriors.py

```diff
@@ -733,65 +764,158 @@ def getMaxMaxEntScanScoreSlidingWindowSNS(variant, windowSize=9, exonicPortionSi
+    Returns integer 1-stdAccSize based on variant position in highest scoring window if donor=False
+    stdDonorSize default value = 9, stdAccSize default value = 23
+    deNovo=True if accounting for deNovoAcceptors, False otherwise
+    accDonor=True if looking for deNovoDonor in ref acceptor site, False otherwise
+    '''
+    slidingWindowInfo = getMaxMaxEntScanScoreSlidingWindowSNS(variant, stdExonicPortion, stdDeNovoLength,
+                                                              donor=donor, deNovo=deNovo, accDonor=accDonor)
     varWindowPos = slidingWindowInfo["varWindowPosition"]
     return varWindowPos
 
-def getSubsequentDonorScores(variant):
+def getClosestSpliceSiteScores(variant, deNovoOffset, donor=True, deNovo=False, accDonor=False):
     '''
-    Given a variant, checks if variant is in an exon
+    Given a variant, determines scores for closest reference splice sequence
+    deNovoOffset refers to difference between de novo acceptor length and exonic portion size
+        default value is 7 (deNovoLength = 10, exonicPortionSize=3)
+       If donor = True, looks for closest splice donor sequence
+       If donor = False, looks for closest splice acceptor sequence
+       If deNovo = True, accomodates for de novo splice acceptors
     If exonic variant, returns a dictionary containing:
-       MaxEntScan score and z-score for reference subsequent donor sequence
+       MaxEntScan score and z-score for reference closest splice sequence
+    If variant located in referene splice site, returns a dictionary containing:
+       MaxEntScan score and z-score for that reference splice site sequence
+    accDonor = False if NOT checking for de novo splice donor sites in reference splice acceptor sites
+    accDonor = True if checking for de novo splice donor sites in reference splice acceptor sites 
     '''
     varGenPos = int(variant["Pos"])
-    inExon = varInExon(variant)
-    if inExon == True:
-        varChrom = getVarChrom(variant)
-        varStrand = getVarStrand(variant)
+    varChrom = getVarChrom(variant)
+    if varInExon(variant) == True and deNovo == False:
         exonNumber = getVarExonNumberSNS(variant)
-        refSpliceDonorBounds = getRefSpliceDonorBoundaries(variant)
-        subDonorBounds = refSpliceDonorBounds[exonNumber]
-        refSeq = getFastaSeq(varChrom, subDonorBounds["donorStart"], subDonorBounds["donorEnd"])
-        subMaxEntScanScore = runMaxEntScan(refSeq, donor=True)
-        subZScore = getZScore(subMaxEntScanScore, donor=True)
-        return {"maxEntScanScore": subMaxEntScanScore,
-                "zScore": subZScore}
+        if donor == True:
+            refSpliceDonorBounds = getRefSpliceDonorBoundaries(variant)
+            closestSpliceBounds = refSpliceDonorBounds[exonNumber]
+        else:
+            refSpliceAccBounds = getSpliceAcceptorBoundaries(variant, deNovo=deNovo)
+            cloesetSpliceBounds = refSpliceAccBounds[exonNumber]
```

Comment:
I didn't write unittests for getClosestSpliceBounds, so that's why misspelling did not cause any errors. I can write unittests to go up in the next commit